### DISCUSSION TITLE
How to work with `auth-proxy` in previews and test function locally

### DISCUSSION THREAD

**Discussion by vadimshvetsov** [t3-oss/create-t3-turbo#1049] [Q&A] [answered]
Hey,

I've tried to deploy auth-proxy with zero config and can't make it works, all envs set correctly according to the README, custom domain added but doesn't work even locally.

I was trying to debug locally and propagated my env using `with-env` script and still can't match the action.

From Vercel logs:
```
[31m[auth][error][0m UnknownAction: Cannot parse action at /callback/google. Read more at https://errors.authjs.dev#unknownaction
```

From localhost logs:
```
[auth][error] UnknownAction: Cannot parse action at /callback/google. Read more at https://errors.authjs.dev#unknownaction at parseActionAndProviderId
```

My `auth-proxy/routes/[...auth].ts`:
```
import { Auth } from "@auth/core"
import Google from "@auth/core/providers/google"
import Resend from "@auth/core/providers/resend"
import { eventHandler, toWebRequest } from "h3"

export default eventHandler(async (event) =>
  Auth(toWebRequest(event), {
    secret: process.env.AUTH_SECRET,
    trustHost: true,
    redirectProxyUrl: process.env.AUTH_REDIRECT_PROXY_URL,
    providers: [
      Google({
        clientId: process.env.GOOGLE_ID,
        clientSecret: process.env.GOOGLE_SECRET,
      }),
      Resend({
        apiKey: process.env.AUTH_RESEND_KEY,
      }),
    ],
  }),
)
```

Appreciate any help.

**Comment by juliusmarminge** [comment, marked as answer]
Try updating to the current state. I added a base path which seems to have resolved some issues

**Reply by vadimshvetsov** [reply]
Thanks, after update it works. Appreciate your efforts 🙏

**Comment by jpainam** [comment]
Hi @vadimshvetsov  How did to run `auth-proxy` locally? I'm using a `GoogeProvider` and `CredentialsProvider`.
Can't run `pnpm dev` from `auth-proxy`. the `script` is available in my `auth-proxy/package.json`

**Reply by vadimshvetsov** [reply]
I didn't try to run it locally, the main point is to run it on the remote server to have stable url for providers

**Reply by juliusmarminge** [reply]
Yea the proxy has no use locally. You want that on a server somewhere with a stable URL

**Comment by jpainam** [comment]
Signing from expo open nextjs/login page and then i see a 303  in the console, then nothing else happen. 

Is there a way to write my my own expo/login page and then call the auth-proxy with the credentials ?
Currently it opens a browser. Not a good user experience IMO

**Comment by jpainam** [comment]
Hi. I was unable to deploy `next-auth` in vercel. This is what i get
```ts
Build Failed
The file "/vercel/path0/apps/auth-proxy/.output/routes-manifest.json" couldn't be found. This is often caused by a misconfiguration in your project.
```

Build command: `turbo run build`
Output Directory: `.output`
Install command: `pnpm install`

Include files outside the root directory was checked

**Reply by juliusmarminge** [reply]
Did you set the NITRO_PRESET variable? https://nitro.unjs.io/deploy/providers/vercel#vercel-edge-functions

**Reply by jpainam** [reply]
I did, i set it to `NITRO_PRESET="vercel_edge"`
